src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleObject.cs

```diff
@@ -17,6 +17,12 @@ public static partial class JsonSerializer
             // Write the start.
             if (!state.Current.StartObjectWritten)
             {
+                if (state.Current.CurrentValue == null)
+                {
+                    state.Current.WriteObjectOrArrayStart(ClassType.Object, writer, writeNull: true);
```

Comment:
Why doesn't this break tests and force-write null when IgnoreNullValues is true (for normal properties that return an object)?

Reply:
It's because the code-flow only reaches here when serializing a root object or a value not belonging to an object e.g. a dictionary value. In these cases we shouldn't ignore `null` values.

I've added a comment and a test verifying that we ignore `null` when serializing nested object properties when `IgnoreNullValues` is true.